hack/csv-generator.go

```diff
@@ -0,0 +1,238 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/blang/semver"
+	gyaml "github.com/ghodss/yaml"
+	"github.com/operator-framework/api/pkg/lib/version"
+	csvv1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	extv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+type generatorFlags struct {
+	dumpCRDs          bool
+	csvVersion        string
+	namespace         string
+	operatorVersion   string
+	validatorImage    string
+	kvmInfoImage      string
+	virtLauncher      string
+	nodeLabellerImage string
+	cpuPlugin         string
+	operatorImage     string
+}
+
+var (
+	f       generatorFlags
+	rootCmd = &cobra.Command{
+		Use:   "csv-generator",
+		Short: "csv-generator for ssp operator",
+		Long:  `csv-generator generates deploy manifest for ssp operator`,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := runGenerator()
+			if err != nil {
+				fmt.Println(err.Error())
```

Comment:
It may be better to print the error to `stderr`, because `stdout` is used for output of generated resources. It would be strange if there was an error message in the middle of valid resources.